tensornetwork/linalg/operations.py

```diff
@@ -0,0 +1,405 @@
+# Copyright 2019 The TensorNetwork Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Union, Text, Optional, List, Sequence, Tuple
+from tensornetwork.tensor import Tensor
+from tensornetwork import ncon_interface
+
+
+def _check_backends(tensors: Sequence[Tensor], fname: str) -> Tuple[bool, str]:
+  """ Checks that each of tensors has the same backend, returning True and an
+      empty string if so, or False and an error string if not.
+  Args:
+    tensors: The list of tensors whose backends to check.
+    fname: The name of the calling function, which will go into the errstring.
+  Returns:
+    (flag, errstr): Whether all backends agree, and an error message if not.
+  """
+  backend = tensors[0].backend
+  backend_names = [tensor.backend.name for tensor in tensors]
+  all_backends_same = True
+  for name in backend_names[1:]:
+    all_backends_same = all_backends_same and name == backend.name
+  errstr = ""
+  if not all_backends_same:
+    errstr = "All Tensors fed to " + fname + "must have the same backend."
+    errstr += "Backends were: \n"
+    errstr += str([name + "\n" for name in backend_names])
+  return all_backends_same, errstr
+
+
+def tensordot(a: Tensor, b: Tensor, axes: Sequence[Sequence[int]]) -> Tensor:
```

Comment:
can axes be an integer (allowed by numpy)?

Reply:
This seems to be fine, and I've changed the type signature accordingly.